### ISSUE TITLE
Number formatting: fractional digits not working for IRIS

### ISSUE THREAD

**Issue by mackers2** [dbeaver/dbeaver#14578]
#### System information: 
- Windows 10 Pro 19043.1348
- DBeaver version 21.2.5
- Additional extensions 

#### Connection specification:
- IRIS 
- Driver name .jdbc.IRISDriver or the dBeaver ODBC driver for ODBC connection
- Do you use tunnels or proxies (SSH, SOCKS, etc)?   No

#### Describe the problem you're observing:
Same as issue reported in https://github.com/dbeaver/dbeaver/issues/13309 for PostgreSQL.

When viewing numeric data in query from IRIS database, the number of digits after the decimal point is only 3, even when fractional digits setting is set to 15.  

The same workaround (with the same limitations) works as in the PostgreSQL issue - disable numeric formatting.

#### Steps to reproduce, if exist:
write select query to show data from table with fields with multiple decimal places
query does not show al decimal places, just the first 3 when numeric formatting set to show 15 fractional digits

**Comment by Matvey16** [comment]
@mackers2 Could you show me your preferences for data formatting, both global and IRIS data source?

**Comment by mackers2** [comment]
I only have global settings - I only started using dBeaver in the last couple of weeks so don't know my way around it well yet.
![image](https://user-images.githubusercontent.com/94457559/142612270-cc66c060-8346-4a5d-ac79-681e57fc730e.png)

**Comment by LonwoLonwo** [comment]
Hello @mackers2 

An example of table DDL will be also very helpful.

**Comment by uslss** [comment]
There has been no update on the issue for a long time.
If it is still present, feel free to reopen it.

**Comment by mackers2** [comment]
Global formatting:
![image](https://user-images.githubusercontent.com/94457559/157277193-83d0ccbe-51dd-4d6d-8c10-032bac4ab75c.png)

IRIS formatting:
![image](https://user-images.githubusercontent.com/94457559/157277284-f8d35349-eaf1-4f5c-ab1b-7db1a5918b57.png)

Display when this format is active:
![image](https://user-images.githubusercontent.com/94457559/157277395-050b2176-0c70-4c92-8860-b1b32f0968ef.png)

(so that's the fault - truncated to 3 decimal places)

Display when numeric formating is disabled:
![image](https://user-images.githubusercontent.com/94457559/157277545-83064f23-6dbc-4986-a7ec-52b1e2e84c0f.png)

These are screenshots from 
![image](https://user-images.githubusercontent.com/94457559/157277731-7dd5a062-7f4e-46cd-971a-de58fe0e48c7.png)

**Comment by mackers2** [comment]
@uslss  please can you reopen this now I've added more info.  I don't seem to be able to.

**Comment by LonwoLonwo** [comment]
Hello @mackers2

What data type has your column with incorrect formatting?

**Comment by mackers2** [comment]
> Hello @mackers2
> 
> What data type has your column with incorrect formatting?


Datatype: %Library.Float
XDBC Type: DOUBLE

is that what you need?

**Comment by LonwoLonwo** [comment]
Hello @mackers2 

Maybe enabling of this setting can help?

![2022-03-13 18_40_05-DBeaver Ultimate 22 0 0 - employee](https://user-images.githubusercontent.com/45152336/158067494-56b67f88-67de-4c26-9377-6ece442490df.png)

**Comment by mackers2** [comment]
> Hello @mackers2
> 
> Maybe enabling of this setting can help?
> 
> ![2022-03-13 18_40_05-DBeaver Ultimate 22 0 0 - employee](https://user-images.githubusercontent.com/45152336/158067494-56b67f88-67de-4c26-9377-6ece442490df.png)

yes that does do what it say (when I logged this I found a similar issue already logged about PostgreSQL:

> Same as issue reported in https://github.com/dbeaver/dbeaver/issues/13309 for PostgreSQL.


> When viewing numeric data in query `from` IRIS database, the number of digits after the decimal point is only 3, even when fractional digits setting is set to 15.


> The same workaround (with the same limitations) works as in the PostgreSQL issue - disable numeric formatting.

**Comment by LonwoLonwo** [comment]
Thanks for the answer @mackers2 

Can I close the ticket?

**Comment by mackers2** [comment]
> Thanks for the answer @mackers2
> 
> Can I close the ticket?

Please can it be fixed as it was for PostgreSQL or is that not possible?

**Comment by LonwoLonwo** [comment]
We read PostgreSQL data types and much more information directly from a database. And we have many PG test environments. Unfortunately, we do not have an IRIS test environment. Perhaps someday, the test environment will appear, and we can do something.

**Comment by mackers2** [comment]
> We read PostgreSQL data types and much more information directly from a database. And we have many PG test environments. Unfortunately, we do not have an IRIS test environment. Perhaps someday, the test environment will appear, and we can do something.

OK I see.  I work for Intersystems and asked a colleague who advised that there is a free version here:

https://community.intersystems.com/post/intersystems-iris-community-edition-now-available-docker-store

Docker link is:
https://hub.docker.com/_/intersystems-iris-data-platform
